### DISCUSSION TITLE
How to express whether the judgment is 0?

### DISCUSSION THREAD

**Discussion by wendy444444** [yalmip/YALMIP#1132] [General discussion] [answered]
I want to make Epro_t(i,1)~=0, Er_t(i,1)=1, Epro_t(i,1) = 0, Er_t(i,1)=0, so I use implies,
`      C = [C,(implies(Epro_t(i,1)~=0,Er_t(i,1)==1)):'Epro_t(i,t)~=1'];
      C = [C,(implies(Epro_t(i,1)==0,Er_t(i,1)==0)):'Epro_t(i,t)==0'];
`
But the result ignored my program and didn't get the desired result.
 Excuse me, is my program correct? If it is wrong, how can it be modified?

**Comment by johanlofberg** [comment, marked as answer]
Implementing ~=0 is numerically horrible, you will have to do it manually with reasonable margins in your context

you should introduce new binaries a, b and c,and use 

```
implies(a,[x == 0,y==0])% Zero
implies(b, [x>= margin, y == 1])% Pos
impllies(c, [x<= -margin), y == 1]) % Neg
a+b+c==1
```

where margin makes sense in your world, and doesn't drown in the general tolerances of numerical solvers (i.e. below 1e-8 makes no sense typically, and the larger the better numerically)

i.e., once again standard application of the general rules to flatten everything and enumerate cases
https://yalmip.github.io/modellingif/

**Comment by wendy444444** [comment]
Thank you sir, it worked.
I would like to ask another question: If I use matlab+yalmip+gurobi to solve the problem, when I encounter different data, but the model and constraints are the same, can I establish a function to represent the model and constraints, and then call this function with different data? If so, how to call the solver? If not, how can we solve this problem?

**Reply by johanlofberg** [reply]
https://yalmip.github.io/command/optimizer/

**Reply by wendy444444** [reply]
Sir, if I have many matrix inputs and outputs, can I use the method linked above?
Do you have any suggestions if I want to shorten the solution time as much as possible?

**Reply by johanlofberg** [reply]
https://yalmip.github.io/tags/#optimizer

**Reply by wendy444444** [reply]
Thank you. I'll try.
